test/source/TestRingBuffer.cpp

```diff
@@ -549,6 +549,31 @@ int TestRingBuffer()
 		EATEST_VERIFY(rbVectorInt[5] == 4);
 	}
 
+	{
+		// Comparation operator ==, operator <
+		// Fix bug mentioned in https://github.com/electronicarts/EASTL/issues/511
+		typedef ring_buffer<int, vector<int>> RBVectorInt;
+
+		RBVectorInt rbVectorInt(3);
+
+		rbVectorInt.push_back(0);
+		rbVectorInt.push_back(1);
+		rbVectorInt.push_back(2);
+		rbVectorInt.push_back(3);
+		rbVectorInt.push_back(4);
+
+		EATEST_VERIFY(rbVectorInt[0] == 2);
+		EATEST_VERIFY(rbVectorInt[1] == 3);
+		EATEST_VERIFY(rbVectorInt[2] == 4);
+
+		RBVectorInt rbVectorInt2({3, 4, 2});
+		RBVectorInt rbVectorInt3({2, 3, 4});
+		RBVectorInt rbVectorInt4({3, 4, 5});
+
+		EATEST_VERIFY(rbVectorInt != rbVectorInt2);
+		EATEST_VERIFY(rbVectorInt == rbVectorInt3);
+		EATEST_VERIFY(rbVectorInt < rbVectorInt4);
```

Comment:
Could you add some tests with ring buffers of different sizes? Both for equality and comparison. And add tests for comparing in the opposite direction as well.

Reply:
Sure. To keep test minimal and simple, I just add one more var `rbVectorInt5` which has different size.

Reply:
I disagree that tests should be minimal and simple. Tests should be as complicated as they need to be ;).

This new test is good but we're still missing a test where one container is a suffix of the other. so comparing `rbVectorInt` against `{2, 3}` or `{2, 3, 4, 5}` for example, and a test where the size is smaller but the lexicographical order is greater, like against `{3, 2}`.

Reply:
Aye Captain, additional necessary test cases have been equipped.